includes/course-theme/class-sensei-course-theme-editor.php

```diff
@@ -0,0 +1,372 @@
+				'<span class="awaiting-mod">' . __( 'beta', 'sensei-lms' ) . '</span>'
+			),
+			'edit_theme_options',
+			'site-editor.php?postType=wp_template'
+		);
+
+	}
+
+
+	/**
+	 * Add template editing hooks for site editor and related API requests.
+	 *
+	 * @access private
+	 */
+	public function maybe_add_site_editor_hooks() {
+
+		$uri = isset( $_SERVER['REQUEST_URI'] ) ? esc_url_raw( wp_unslash( $_SERVER['REQUEST_URI'] ) ) : '';
+
+		$is_site_editor      = preg_match( '#/wp-admin/site-editor.php#i', $uri );
+		$is_site_editor_rest = preg_match( '#/wp-json/.*/' . self::THEME_PREFIX . '#i', $uri ) || preg_match( '#/wp-json/wp/v2/templates#i', $uri );
+
+		if ( $is_site_editor || $is_site_editor_rest ) {
+
+			$this->add_site_editor_hooks();
+
+			if ( ! function_exists( 'wp_is_block_theme' ) || ! wp_is_block_theme() && ! Sensei_Course_Theme::instance()->is_active() ) {
+				Sensei_Course_Theme::instance()->override_theme();
+			}
+		}
+
+	}
+
+	/**
+	 * Add template editing hooks.
+	 *
+	 * @access private
+	 */
+	public function add_site_editor_hooks() {
+
+		add_action( 'enqueue_block_editor_assets', [ $this, 'enqueue_site_editor_assets' ] );
+		add_action( 'admin_init', [ $this, 'add_editor_styles' ] );
+		add_filter( 'get_block_templates', [ $this, 'add_course_theme_block_templates' ], 10, 3 );
+		add_filter( 'pre_get_block_file_template', array( $this, 'get_single_block_template' ), 10, 3 );
+	}
+
+	/**
+	 * Return a block template for the course theme.
+	 *
+	 * @param \WP_Block_Template|null $template      Return a block template object to short-circuit the default query,
+	 *                                               or null to allow WP to run its normal queries.
+	 * @param string                  $id            Template unique identifier (example: theme_slug//template_slug).
+	 * @param array                   $template_type wp_template or wp_template_part.
+	 *
+	 * @return mixed|\WP_Block_Template|\WP_Error
+	 */
+	public function get_single_block_template( $template, $id, $template_type ) {
+
+		if (
+			'wp_template' !== $template_type || ! $id || 0 !== strpos( $id, self::THEME_PREFIX . '//' )
+		) {
+			return $template;
+		}
+
+		$templates = $this->get_block_templates();
+
+		list( , $slug ) = explode( '//', $id );
+
+		return $templates[ $slug ] ?? $template;
+	}
+
+	/**
+	 * Register block templates as plugin-provided templates when a block theme is already active.
+	 *
+	 * @access private
+	 */
+	public function register_plugin_templates() {
+	}
+
+	/**
+	 * Enqueue course theme blocks and styles.
+	 *
+	 * @access private
+	 */
+	public function enqueue_site_editor_assets() {
+		Sensei()->assets->enqueue( Sensei_Course_Theme::THEME_NAME . '-style', 'css/sensei-course-theme.css' );
+		Sensei()->assets->enqueue( Sensei_Course_Theme::THEME_NAME . '-editor-style', 'css/sensei-course-theme.editor.css' );
+		Sensei()->assets->enqueue( Sensei_Course_Theme::THEME_NAME . '-blocks', 'course-theme/blocks.js' );
```

Comment:
I'm not seeing the warnings, is it in JS? `Sensei_Blocks` should be registering the category in the backend, and `registerSenseiBlocks` in the editor.

Reply:
Yep!

In the JS:
<img width="1471" alt="Screen Shot 2022-01-18 at 18 17 04" src="https://user-images.githubusercontent.com/876340/150020837-e4a78b6e-75e0-4339-85c0-7b560aaadfb4.png">

I was checking that the `Sensei_Blocks::sensei_block_categories`contains a check that doesn't fit with this case. I think we have to update that:

```php
if ( ! $post || ! in_array( $post->post_type, [ 'course', 'lesson', 'question', 'page' ], true ) ) {
```

Reply:
Oh thanks for looking into it. Removing that check in 6963982415d146129e5ca6cf0218e74c50e0b4cc, because looks like the category won't show up if there are no blocks in it, so it should be safe to always register it.